client/components/PrettyPrint/PrettyPrint.jsx

```diff
@@ -0,0 +1,76 @@
+import { h, Component } from "preact";
+import cx from "classnames";
+import Prism from "prismjs";
+
+// Import prism plugins
+import "prismjs/plugins/line-numbers/prism-line-numbers";
+import "prismjs/plugins/line-highlight/prism-line-highlight";
+
+import "./prettyprint.css";
+
+const BASE_CLASS = "pretty-print";
+
+export default class PrettyPrint extends Component {
+  constructor(...args) {
+    super(...args);
+    this.highlight = this.highlight.bind(this);
```

Comment:
If I don't do this it not have access to `props` and `state`. It does not auto-bind.